---
title: "scrna-seq-analysis"
author: "Victoria Latynina"
date: "2023-03-18"
output:
  pdf_document: default
  html_document: default
---

## Single Cell RNA-seq analysis

### Pipeline steps:



```{r rna, warning=F, message=F}
library(Seurat)
library(Matrix)
library(MAST)
library(ggplot2)
library(dplyr)
```

## Combine counts into count matrix

```{r rna1, warning=F, message=F}
data <- Read10X("~/Code/sys_bio/Day 5 - March 22th/scRNA-seq/materials/filtered_feature_bc_matrix/")
dim(data)
```

## Creating Seurat object and filtering (genes and barcodes)

```{r rna2, warning=F, message=F}

seurat <- CreateSeuratObject(data, min.cells = 20, min.features = 20)
dim(seurat)
FeatureScatter(seurat, "nCount_RNA", "nFeature_RNA") + scale_x_log10() + scale_y_log10()
# Get and plot mitochondrial content in cells
seurat[["percent.mt"]] <- PercentageFeatureSet(seurat, pattern = "^MT-")
FeatureScatter(seurat, "nCount_RNA", "percent.mt") + scale_x_log10()
FeatureScatter(seurat, "nFeature_RNA", "percent.mt") + scale_x_log10()

seurat <- subset(seurat, subset = nFeature_RNA > 1000 & percent.mt < 25)
dim(seurat)

```

## Normalize data and get a tSNE plot

```{r rna3, warning=F, message=F}

seurat <- SCTransform(seurat, vars.to.regress = "percent.mt")

VariableFeaturePlot(seurat) + scale_y_log10()

top10_variable_genes <- head(VariableFeatures(seurat), 10)
VariableFeaturePlot(seurat) %>%
  LabelPoints(points = top10_variable_genes, repel = TRUE) +
  scale_y_log10()
# PCA
seurat <- RunPCA(seurat)
ElbowPlot(seurat, ndims = 50)
# TSNE
seurat <- RunTSNE(seurat, dims=1:20)
DimPlot(seurat, reduction = "tsne") + NoLegend()

```

## Get UMAP and tSNE

```{r rna4, warning=F, message=F }
seurat <- RunUMAP(seurat, dims=1:20)
DimPlot(seurat, reduction = "umap") + NoLegend()

DimPlot(seurat, reduction = "tsne") + NoLegend()
DimPlot(seurat, reduction = "umap") + NoLegend()
# Get graph of neighbors using 20 components
seurat <- FindNeighbors(seurat, dims = 1:20)
# Clustering
seurat <- FindClusters(seurat, resolution=0.6)
DimPlot(seurat, reduction = "tsne", label = TRUE) + NoLegend()
DimPlot(seurat, reduction = "umap", label = TRUE) + NoLegend()
```

## Umap

```{r rna567, echo=FALSE, warning=F, message=F}

FeaturePlot(seurat, c("CD14", "CD79A", "CD3D"), cols=c("grey", "red"), reduction="umap", ncol=3)

# max cells per ident is only seed to speed up the whole thing
allMarkers <- FindAllMarkers(seurat, max.cells.per.ident = 100, only.pos = T)
goodMarkers <- allMarkers %>% group_by(cluster) %>% top_n(n = 1, wt = avg_log2FC) %>% pull(gene)
goodMarkers


FeaturePlot(seurat, goodMarkers[1:3], cols=c("grey", "red"), reduction="umap", ncol=3)
VlnPlot(seurat, goodMarkers[1:3], pt.size = 0.1)

```

